src/ucp/api/ucp.h

```diff
@@ -431,6 +470,108 @@ typedef struct ucp_context_attr {
+     */
+    uint64_t                field_mask;
+
+    /**
+     * Thread safety "mode" for the worker object and resources associated with it.
+     * This value is optional.
+     * If it's not set (along with its corresponding bit in the field_mask -
+     * UCP_WORKER_PARAM_FIELD_THREAD_MODE), the UCS_THREAD_MODE_SINGLE mode
+     * will be used.
+     */
+    ucs_thread_mode_t       thread_mode;
+} ucp_worker_params_t;
+
+
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief Tuning parameters for the UCP endpoint.
+ *
+ * The structure defines the parameters that are used for the
+ * UCP endpoint tuning during the UCP ep @ref ucp_ep_create "creation".
+ */
+typedef struct ucp_ep_params {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_ep_params_field.
+     * Fields not specified in this mask would be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t                field_mask;
+
+    /**
+     * Destination address; the address must be obtained using
+     * @ref ucp_worker_get_address.
+     * This field is mandatory for filling (along with its corresponding bit
+     * in the field_mask - UCP_EP_PARAM_FIELD_REMOTE_ADDRESS).
+     * The ucp_ep_create routine will return with an error if the address isn't
+     * specified.
+     */
+    const ucp_address_t     *address;
+} ucp_ep_params_t;
+
+
+/**
+ * @ingroup UCP_MEM
+ * @brief Tuning parameters for the UCP memory mapping.
+ *
+ * The structure defines the parameters that are used for the
+ * UCP memory mapping tuning during the @ref ucp_mem_map "ucp_mem_map" routine.
+ */
+typedef struct ucp_mem_map_params {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_mem_map_params_field.
+     * Fields not specified in this mask would be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t                field_mask;
+
+    /**
+     * If the address is not NULL, the routine maps (registers) the memory segment
+     * pointed to by this address.
+     * If the pointer is NULL, the library allocates mapped (registered) memory
+     * segment and returns its address in this argument.
+     * Therefore, this value is optional.
+     * If it's not set (along with its corresponding bit in the field_mask -
+     * UCP_MEM_MAP_PARAM_FIELD_ADDRESS), the ucp_mem_map routine will consider
+     * address as set to NULL and will allocate memory.
+     */
+     void                   *address;
+
+     /**
+      * Length (in bytes) to allocate or map (register).
+      * This field is mandatory for filling (along with its corresponding bit
+      * in the field_mask - UCP_MEM_MAP_PARAM_FIELD_LENGTH).
+      * The ucp_mem_map routine will return with an error if the length isn't
+      * specified.
+      */
+     size_t                 length;
+
+     /**
+      * Allocation flags, UCP_MEM_MAP_xx.
```

Comment:
Okay. Please reflect this in documentation (XX does not help much). I would suggest use @ref to the enum